lib/tasks_private/spec_helper.rake

```diff
@@ -4,15 +4,47 @@ namespace :spec do
     tower_host = ENV['TOWER_URL'] || "https://dev-ansible-tower3.example.com/api/v1/"
     id = ENV['TOWER_USER'] || 'testuser'
     password = ENV['TOWER_PASSWORD'] || 'secret'
-    PopulateTower.new(tower_host, id, password).create_dataset.counts
+    PopulateTower.new(tower_host, id, password, TowerData.new(true)).create_dataset.counts
   end
 
   desc "Get counts of various Tower objects"
   task :tower_counts do
     tower_host = ENV['TOWER_URL'] || "https://dev-ansible-tower3.example.com/api/v1/"
     id = ENV['TOWER_USER'] || 'testuser'
     password = ENV['TOWER_PASSWORD'] || 'secret'
-    PopulateTower.new(tower_host, id, password).counts
+    PopulateTower.new(tower_host, id, password, TowerData.new(false)).counts
+  end
+end
+
+class TowerData
```

Comment:
This is pretty cool.  However, it's overkill in this case and that introduces unnecessary mental weight for forthcoming maintainers.  Can we simply dump the collected hash into yml after creating the data set?